shared-bindings/socketpool/SocketPool.c

```diff
@@ -83,6 +83,30 @@ MP_DEFINE_EXCEPTION(gaierror, OSError)
 //|
 //|     IPPROTO_TCP: int
 //|     IPPROTO_IP: int
+//|     IPPROTO_AH: int
+//|     IPPROTO_DSTOPTS: int
+//|     IPPROTO_EGP: int
+//|     IPPROTO_ESP: int
+//|     IPPROTO_FRAGMENT: int
+//|     IPPROTO_GRE: int
+//|     IPPROTO_HOPOPTS: int
+//|     IPPROTO_ICMP: int
+//|     IPPROTO_ICMPV6: int
+//|     IPPROTO_IDP: int
+//|     IPPROTO_IGMP: int
+//|     IPPROTO_IPIP: int
+//|     IPPROTO_IPV6: int
+//|     IPPROTO_MPTCP: int
+//|     IPPROTO_NONE: int
+//|     IPPROTO_PIM: int
+//|     IPPROTO_PUP: int
+//|     IPPROTO_RAW: int
+//|     IPPROTO_ROUTING: int
+//|     IPPROTO_RSVP: int
+//|     IPPROTO_SCTP: int
+//|     IPPROTO_TP: int
+//|     IPPROTO_UDP: int
+//|     IPPROTO_UDPLITE: int
```

Comment:
Adding them here does not make them available in Python You need to add them around line 221 in this file. But there are too many to add, and most will not be used. Just add a few of the common ones, including ICMP.

For instance, MicroPython defines these common ones:
https://github.com/micropython/micropython/blob/9feb0689eeaca5ce88aedcc680f997a3b4d0221c/extmod/modsocket.c#L640

Reply:
IPPROTO_ICMP=1 and IPPROTO_RAW=255 are the 2 that'd I'd recommend explicitly adding to those that are already implemented in CP.